### DISCUSSION TITLE
Showing full referrer url?

### DISCUSSION THREAD

**Discussion by Brawl345** [umami-software/umami#2995] [Q&A]
Can I somehow see the full referrer URL and not just the domain? I can see them while querying in the database but not in the UI. I'm of course looking at websites that send the full referrer path.

**Comment by DannyBen** [comment]
I wonder why is this not addressed? Seems like an important bug.

**Comment by dansku** [comment]
100% agree!

**Comment by mklnz** [comment]
Is this data recorded at all? I guess I can check in the DB

**Reply by ddlaws0n** [reply]
So… was it in the db after all?

**Reply by Brawl345** [reply]
It is (that's why I opened this issue), I use this query:

```sql
SELECT we.created_at, w.name, CONCAT(we.referrer_domain, we.referrer_path) AS referrer FROM website_event we JOIN website w ON w.website_id = we.website_id WHERE referrer_domain NOT LIKE '%example.com%' AND referrer_path NOT IN('#', '/', '/link', '/s', '/search', '/url') ORDER BY created_at DESC;
```

The first "NOT" query filters out referrers from my own domain (replace example.com obviously) and the second one (referrer_path) filters out base domains and other junk I don't care about, you can also leave that out. Then you can see the full url. Maybe you can also filter which page was visited by which referrer and make it distinct, didn't care about that, only wanted to see from where people come from exactly.

**Comment by securityopa** [comment]
any resolutions on this ?

**Comment by dansku** [comment]
actually this will never work, the browser doesn't send this data. so there's no way for umami to fetch it. i'd close it

**Reply by Brawl345** [reply]
Totally wrong and disproved if you'd have read the thread.

**Reply by mklnz** [reply]
No he is right, only the domain is retained and this is a limitation of the browser. If you look in the DB any external referrer domain has the referrer_path of "/".

Please see: https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Referrer-Policy

The default setting for most browsers is: strict-origin-when-cross-origin

**Reply by Brawl345** [reply]
No this is wrong, I see the full path there myself and the SQL statement I shared above already filters the `/` path. This is not a browser limitation, some sites (many) suppress sending the referer via the Referrer-Policy header or send no header at all but not every site does it. Bitly still sends the full path for example, as does Reddit and GitHub.

Fact is that the path is already in the DB and it would be nice if Umami could show it.

**Reply by DannyBen** [reply]
I think the main thing to realize here, is that it does not matter if it is the default browser behavior, or if many or few sites send that full path. The thing is - if it exists, lets show it. The real question is - why does Umami feel the need to NOT show it? I mean, this was probably a deliberate action, no? Probably due to a common misconception or partial evidence, so I mean no disrespect here to the wonderful work done for free for the benefit of all mankind - but why not just undo this "trimming", even if in a draft release, for the people in this thread to see if it helps them or not.

I can also understand it if the owners just say "wont fix" and we can all move on.

**Reply by mklnz** [reply]
@DannyBen I guess that's fair, can't disagree with you there
